### PR TITLE
Handle errors properly in the extracted encoder

### PR DESCRIPTION
**Pull request by jerzywilczek**

### CODE REVIEW

File: compositor_pipeline/src/pipeline.rs [hunk 1]
```diff
@@ -58,14 +62,22 @@ pub struct PipelineOutput<Receiver: PipelineOutputReceiver> {
...
         std::thread::spawn(move || {
-            let mut receiver = Receiver::new(opts.receiver_options, encoder.codec());
-            id_tx.send(receiver.identifier()).unwrap();
+            let mut receiver = match Receiver::new(opts.receiver_options, encoder.codec()) {
+                Ok(r) => r,
+                Err(e) => {
+                    id_tx.send(Err(e)).unwrap();
+                    return;
+                }
+            };
```
Comment (WojciechBarczynski) [inline compositor_pipeline/src/pipeline.rs:77]: `unwrap_or_else`

**Review by noituri — APPROVED** [review]

File: compositor_pipeline/src/pipeline/encoder.rs [hunk 2]
```diff
@@ -83,56 +85,49 @@ pub(crate) struct Encoder {
...
-                    // QP factor between P and B frames - libx264 defaults to 1.4 (in case of tune=grain to 1.1)
-                    ("f_pb_factor", "1.3"),
-                    // A comma-separated list of partitions to consider. Possible values: p8x8, p4x4, b8x8, i8x8, i4x4, none, all
-                    ("partitions", settings.preset.default_partitions()),
-                    // Subpixel motion estimation and mode decision (decision quality: 1=fast, 11=best)
-                    ("subq", settings.preset.default_subq_mode()),
-                ]),
-            )
-            .map_err(|e| e.to_string())?;
+        let encoder = encoder.open_as_with(
```
Comment (wkozyra95) [inline compositor_pipeline/src/pipeline/encoder.rs:98]: Perfect, now you are going to be in a `git blame` for that :D

File: compositor_pipeline/src/pipeline.rs (hunk 1 above)
Comment (wkozyra95) [reply compositor_pipeline/src/pipeline.rs:77]: this is unwrap on sending to channel, it can only happen if other thread panics and mutex is posioned

**Review by wkozyra95 — APPROVED** [review]

File: compositor_pipeline/src/pipeline.rs (hunk 1 above)
Comment (WojciechBarczynski) [reply compositor_pipeline/src/pipeline.rs:77]: Nevermind, haven't thought about return

**Review by WojciechBarczynski — APPROVED** [review]

**Pull request merged**